src/libs/auth.js

```diff
@@ -21,6 +21,28 @@ export const reloadAuthToken = () => {
   return getAuthInstance().currentUser.get().reloadAuthResponse().then(() => true, () => false)
 }
 
+export const hasBillingScope = () => {
+  return getAuthInstance().currentUser.get().hasGrantedScopes('https://www.googleapis.com/auth/cloud-billing')
+}
+
+/*
+ * Request Google Cloud Billing scope if necessary, then do the thing.
+ *
+ * NOTE: Requesting additional scopes may invoke a browser pop-up which the browser might block.
+ * If you use withBillingScope during page load and the pop-up is blocked, onFailure will be called.
+ * In this case, you'll need to provide something for the user to deliberately click on and retry
+ * withBillingScope in reaction to the click.
+ */
+export const withBillingScope = (f, onFailure) => {
```

Comment:
My Scala is showing a little bit here. Let me know if there's a better way to express this pattern.

Reply:
This could just be an async function, e.g. `ensureBillingScope`. If we have the scope already, just return. Otherwise, await on the google stuff, and then return. Then the consumer can just do `await ensureBillingScope()`